src/core/math.d

```diff
@@ -164,3 +164,101 @@ unittest
     }
 }
 
+/*************************************
+ * Round argument to float precision.
+ *
+ * D language types specify a minimum precision, not
+ * a maximum. toFloat() forces float precision.
+ * Params:
+ *      f = value to convert
+ * Returns:
+ *      f in float precision
+ */
+
+@safe pure nothrow
+float toFloat(float f)  { pragma(inline, false); return f; }
+
+/// ditto
+@safe pure nothrow
+float toFloat(double f) { pragma(inline, false); return cast(float)f; }
+
+/// ditto
+@safe pure nothrow
+float toFloat(real f)   { pragma(inline, false); return cast(float)f; }
+
+@safe unittest
+{
+    static float f = 1.1f;
+    static double d = 1.1;
+    static real r = 1.1L;
+    f = toFloat(f + f);
+    f = toFloat(d + d);
+    f = toFloat(r + r);
+}
```

Comment:
I initially tried that, but got different results on different platforms :-(. The compiler needs to recognize them as intrinsics to get it consistent.